app/Filament/Resources/NodeResource/RelationManagers/AllocationsRelationManager.php

```diff
@@ -97,18 +99,26 @@ public function table(Table $table): Table
                             ->label('Ports')
                             ->inlineLabel()
                             ->live()
-                            ->afterStateUpdated(function ($state, Set $set) {
+                            ->afterStateUpdated(function ($state, Set $set, Get $get) {
                                 $ports = collect();
                                 $update = false;
                                 foreach ($state as $portEntry) {
                                     if (!str_contains($portEntry, '-')) {
                                         if (is_numeric($portEntry)) {
-                                            $ports->push((int) $portEntry);
-
-                                            continue;
+                                            if (Allocation::query()->where('ip', $get('allocation_ip'))->where('port', $portEntry)->exists()) {
```

Comment:
This needs to happen after the foreach loop because of the [N+1 problem](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping)!